azurerm/resource_arm_postgresql_server.go

```diff
@@ -0,0 +1,306 @@
+			AdministratorLoginPassword: azure.String(adminLoginPassword),
+			CreateMode:                 postgresql.CreateModeDefault,
+		},
+		Tags: expandTags(tags),
+	}
+
+	_, error := client.CreateOrUpdate(resGroup, name, properties, make(chan struct{}))
+	err := <-error
+	if err != nil {
+		return err
+	}
+
+	read, err := client.Get(resGroup, name)
+	if err != nil {
+		return err
+	}
+	if read.ID == nil {
+		return fmt.Errorf("Cannot read PostgreSQL Server %s (resource group %s) ID", name, resGroup)
+	}
+
+	d.SetId(*read.ID)
+
+	return resourceArmPostgreSQLServerRead(d, meta)
+}
+
+func resourceArmPostgreSQLServerUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).postgresqlServersClient
+
+	log.Printf("[INFO] preparing arguments for AzureRM PostgreSQL Server update.")
+
+	name := d.Get("name").(string)
+	resGroup := d.Get("resource_group_name").(string)
+
+	adminLoginPassword := d.Get("administrator_login_password").(string)
+	sslEnforcement := d.Get("ssl_enforcement").(string)
+	version := d.Get("version").(string)
+	storageMB := d.Get("storage_mb").(int)
+	sku := expandAzureRmPostgreSQLServerSku(d, storageMB)
+
+	tags := d.Get("tags").(map[string]interface{})
+
+	properties := postgresql.ServerUpdateParameters{
+		Sku: sku,
+		ServerUpdateParametersProperties: &postgresql.ServerUpdateParametersProperties{
+			SslEnforcement:             postgresql.SslEnforcementEnum(sslEnforcement),
+			StorageMB:                  azure.Int64(int64(storageMB)),
+			Version:                    postgresql.ServerVersion(version),
+			AdministratorLoginPassword: azure.String(adminLoginPassword),
+		},
+		Tags: expandTags(tags),
+	}
+
+	_, error := client.Update(resGroup, name, properties, make(chan struct{}))
+	err := <-error
+	if err != nil {
+		return err
+	}
+
+	read, err := client.Get(resGroup, name)
+	if err != nil {
+		return err
+	}
+	if read.ID == nil {
+		return fmt.Errorf("Cannot read PostgreSQL Server %s (resource group %s) ID", name, resGroup)
+	}
+
+	d.SetId(*read.ID)
+
+	return resourceArmPostgreSQLServerRead(d, meta)
+}
+
+func resourceArmPostgreSQLServerRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).postgresqlServersClient
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resGroup := id.ResourceGroup
+	name := id.Path["servers"]
+
+	resp, err := client.Get(resGroup, name)
+	if err != nil {
+		if responseWasNotFound(resp.Response) {
+			d.SetId("")
+			return nil
+		}
+
+		return fmt.Errorf("Error making Read request on Azure PostgreSQL Server %s: %+v", name, err)
+	}
+
+	d.Set("name", resp.Name)
+	d.Set("resource_group_name", resGroup)
+	d.Set("location", azureRMNormalizeLocation(*resp.Location))
+
+	d.Set("administrator_login", resp.AdministratorLogin)
+	d.Set("version", string(resp.Version))
+	d.Set("storage_mb", int(*resp.StorageMB))
+	d.Set("ssl_enforcement", string(resp.SslEnforcement))
+
+	flattenAndSetAzureRmPostgreSQLServerSku(d, resp.Sku)
```

Comment:
Nitpick - The single-responsibility principle says to me this isn't clean :) can't we just let the function do the flattening and then `d.Set()` here inline?

Reply:
Good spot - fixed.